adapters/appnexus/appnexus.go

```diff
@@ -327,43 +329,35 @@ func (a *AppNexusAdapter) MakeRequests(request *openrtb.BidRequest, reqInfo *ada
 	}
 
 	// Add Appnexus request level extension
-	var isAMP int
+	var isAMP, isVIDEO int
 	if reqInfo.PbsEntryPoint == pbsmetrics.ReqTypeAMP {
 		isAMP = 1
-	} else {
-		isAMP = 0
+	} else if reqInfo.PbsEntryPoint == pbsmetrics.ReqTypeVideo {
+		isVIDEO = 1
 	}
+
 	var reqExt appnexusReqExt
 	if len(request.Ext) > 0 {
-		if err := json.Unmarshal(request.Ext, &reqExt); err == nil {
-			includeBrandCategory := reqExt.Prebid.Targeting != nil && reqExt.Prebid.Targeting.IncludeBrandCategory.PrimaryAdServer != 0
-			if includeBrandCategory {
-				if reqExt.Appnexus == nil {
-					reqExt.Appnexus = &appnexusReqExtAppnexus{}
-				}
-				reqExt.Appnexus.BrandCategoryUniqueness = &includeBrandCategory
-				reqExt.Appnexus.IncludeBrandCategory = &includeBrandCategory
-				reqExt.Appnexus.IsAMP = isAMP
-				request.Ext, err = json.Marshal(reqExt)
-				if err != nil {
-					errs = append(errs, err)
-					return nil, errs
-				}
-			}
-		} else {
+		if err := json.Unmarshal(request.Ext, &reqExt); err != nil {
 			errs = append(errs, err)
 			return nil, errs
 		}
-	} else if isAMP != 0 {
-		// if isAMP is 0, bidder doesn't need it included
-		var err error
+	}
+	if reqExt.Appnexus == nil {
```

Comment:
In the previous version of the code, the `if reqExt.Appnexus == nil` condition in line 346 was inside the `if includeBrandCategory` if statement, which made it dependant upon this condition to be true. Whay are we changing this?
```
346     if reqExt.Appnexus == nil {
347         reqExt.Appnexus = &appnexusReqExtAppnexus{}
348     }
349     includeBrandCategory := reqExt.Prebid.Targeting != nil && reqExt.Prebid.Targeting.IncludeBrandCategory.PrimaryAdServer != 0
350     if includeBrandCategory {
351         reqExt.Appnexus.BrandCategoryUniqueness = &includeBrandCategory
352         reqExt.Appnexus.IncludeBrandCategory = &includeBrandCategory
353     }
adapters/appnexus/appnexus.go
```

Reply:
So the regExt.Appnexus was being created before only if it was needed, which was when either the includeBrandCategories or isAMP flags were on. With this hb_source field we always want to send it, so now the regExt.Appnexus object needs to be created all the time even in the case where includeBrandCategories and isAMP are both off.

Reply:
Ohhh I see